docs/using/frameworks/flask.rst

```diff
@@ -0,0 +1,196 @@
+Flask
+=====
+
+The oso Flask integration provides a more convenient interface to oso for
+usage with Flask.
+
+Usage
+-----
+
+Initialization
+~~~~~~~~~~~~~~
+
+The :py:class:`flask_oso.FlaskOso` class is the entrypoint to the integration.
+It must be initialized with the Flask app and oso::
+
+    from flask import Flask
+    from oso import Oso
+
+    app = Flask()
+    oso = Oso()
+    flask_oso = FlaskOso(app=app, oso=oso)
+
+Alternatively, to support the Flask factory pattern, the
+:py:meth:`flask_oso.FlaskOso.init_app` method can be used::
+
+    from flask import Flask
+
+    from oso import Oso
+    from flask_oso import FlaskOso
+
+    oso = Oso()
+    flask_oso = FlaskOso(oso=oso)
+
+    def create_app():
+        app = Flask()
+
+        # Initialize oso for this application
+        flask_oso.init_app(app)
+
+        return app
+
+    app = create_app()
+
+This factory function can be a useful place for loading policy files, and
+calling configuration functions on :py:class:`flask_oso.FlaskOso` like
+:py:meth:`flask_oso.FlaskOso.require_authorization`::
+
+    def create_app():
+        app = Flask()
+
+        flask_oso.init_app(app)
+        flask_oso.require_authorization(app)
+
+        oso.load_file("authorization.polar")
+        oso.register_class(Expense)
+
+Performing authorization
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using the ``flask_oso`` integration, the primary authorization function is
+:py:meth:`flask_oso.FlaskOso.authorize`.  It accepts the same arguments as
+:py:meth:`oso.Oso.is_allowed`, but provides sensible defaults for working with
+Flask. The actor defaults to ``flask.g.current_user`` (this can be
+customized, see :py:meth:`flask_oso.FlaskOso.set_get_actor`).  The ``action``
+defaults to the method of the current request ``flask.request.method``.
+``resource`` must be provided.
+
+.. tip::
+
+    If you aren't familiar with how oso uses actors, actions, and resources to
+    express authorization decisions, see :doc:`/more/glossary` or
+    :doc:`/getting-started/quickstart`.
+
+:py:meth:`flask_oso.FlaskOso.authorize` can be used within route handlers, or in
+the data access layer, depending upon how you want to express authorization.
+Here's a basic example in a route::
+
+    @bp.route("/<int:id>", methods=["GET"])
+    def get_expense(id):
+        expense = Expense.query.get(id)
+        if expense is None:
+            raise NotFound()
+
+        oso.authorize(action="read", resource=expense)
+        return expense.json()
+
+Notice we didn't need to check the return value of ``authorize``.  By default,
+a failed authorization will return a ``403 Forbidden`` response for the current
+request. This can be controlled with
```

Comment:
```suggestion
Notice we didn't need to check the return value of ``authorize``.  **By default,
a failed authorization will return a ``403 Forbidden`` response for the current
request.** This can be controlled with
```